src/reformat/rgb.rs

```diff
@@ -457,4 +543,38 @@ mod tests {
         }
         Ok(())
     }
+
+    #[test]
+    fn channel_shuffling() {
+        let image = Image {
+            width: 1,
+            height: 1,
+            depth: 8,
+            format: Format::Rgba,
+            pixels: Some(Pixels::Buffer(vec![0u8, 1, 2, 3])),
+            row_bytes: 4,
+            ..Default::default()
+        };
+
+        let shuffled = image.shuffle_channels_to(Format::Rgba);
+        match &shuffled.unwrap().pixels {
+            Some(Pixels::Pointer(ptr)) => assert_eq!(
+                unsafe { std::slice::from_raw_parts(*ptr, 4) },
```

Comment:
you can use `shuffled.row(..)` to get rid of this unsafe block.

ditto below. we only care about the actual pixel values in this test and now how they are stored underneath (pointer, buffer or buffer16 does not really matter).